### ISSUE TITLE
Upload- progressbar still runs after upload is finished

### ISSUE THREAD

**Issue by sirtet** [aporto/mariamole#38]
I imported the Blink-example and uploaded it. After about 1sec. the program runs, as the blinking led tells me. But the Progress-Bar with the cancel-button is only at about 10%, and finishes 10sec. later...
This is on a mega 2560 Board.

**Comment by AlexxNB** [comment]
Have you seen "Project Blink binaries successfully uploaded to board" in msgs tab?

**Comment by sirtet** [comment]
No, there's a "Error uploading... ...see output window" message, but the output window has no error messages.

PS:
On an arduino uno this runs well, the success message is there after less than 1sec. uploading.

**Comment by aporto** [comment]
Hi, sirtet. Which board is yours?

**Comment by AlexxNB** [comment]
I have same trouble with Nano aprox in 1 of 20 uploads. Firmware uploads succesfuly, but in msgs tab appears "Error uploading...".

**Comment by aporto** [comment]
One thing that may cause this is that MariaMole is not receiving the
AVRDUDE output, despite the upload is correcly finished. In these cases,
MariaMole will terminate AVRDUDE process, and then return an error message.

Please try increasing the upload timeout (Menu Edit->Preferences) and see
if that changes this behaviour

**Comment by sirtet** [comment]
> Hi, sirtet. Which board is yours?

Arduino Mega 2560

> Please try increasing the upload timeout

Changing the Timeout to 60s fixes the issue, returning to 30s brings it back. Strange, My Upload is 1sec. and then the progress bar finishes 10s later, so still well within the original 30s timeout...

**Comment by aporto** [comment]
Even if avrdude finishes, MariaMole still waits for the operating system to
close the avrdude process. If it stop responding after avrdude has
finished, MariaMole will probably wait for the timeout and request a
process kill, and return an error message, even if the upload has happened
correctly

Gonna need some more time for a better solution

**Comment by AlexxNB** [comment]
Maybe close progressbar by Output message parsing and check avrdude proccess in background?

**Comment by aporto** [comment]
That's a solution... :)

**Comment by AlexxNB** [comment]
Sorry, I deleted my msg after reading your pre-previous post =)
